src/garage/torch/policies/tanh_gaussian_mlp_policy.py

```diff
@@ -0,0 +1,184 @@
+            layer_normalization=layer_normalization)
+
+    def get_action(self, observation):
+        """Get a single action given an observation.
+
+        Args:
+            observation (np.ndarray): Observation from the environment.
+
+        Returns:
+            tuple:
+                * np.ndarray: Predicted action.
+                * dict:
+                    * list[float]: Mean of the distribution
+                    * list[float]: Standard deviation of logarithmic values of
+                        the distribution
+
+        """
+        with torch.no_grad():
+            if not isinstance(observation, torch.Tensor):
+                observation = tu.from_numpy(observation)
+            observation = torch.Tensor(observation).unsqueeze(0)
+            dist = self.forward(observation)
+            ret_mean = dist.mean.squeeze(0).cpu().numpy()
+            ret_log_std = (dist.variance**.5).log().squeeze(0).cpu().numpy()
+            return (dist.rsample().squeeze(0).cpu().numpy(),
+                    dict(mean=ret_mean, log_std=ret_log_std))
+
+    def get_actions(self, observations):
+        """Get actions given observations.
+
+        Args:
+            observations (np.ndarray): Observations from the environment.
+
+        Returns:
+            tuple:
+                * np.ndarray: Predicted actions.
+                * dict:
+                    * list[float]: Mean of the distribution
+                    * list[float]: Standard deviation of logarithmic values of
+                        the distribution
+
+        """
+        with torch.no_grad():
+            if not isinstance(observations, torch.Tensor):
+                observations = tu.from_numpy(observations)
+            dist = self.forward(torch.Tensor(observations))
+            ret_mean = dist.mean.cpu().numpy()
+            ret_log_std = (dist.variance**.5).log().cpu().numpy()
+            return (dist.rsample().cpu().numpy(),
+                    dict(mean=ret_mean, log_std=ret_log_std))
+
+    def log_likelihood(self, observation, action):
+        """Compute log likelihood given observations and action.
+
+        Args:
+            observation (torch.Tensor): Observation from the environment.
+            action (torch.Tensor): Predicted action.
+
+        Returns:
+            torch.Tensor: Calculated log likelihood value of the action given
+                observation
+
+        """
+        dist = self.forward(observation)
+        return dist.log_prob(action)
+
+    def entropy(self, observation):
+        """Get entropy given observations.
+
+        Args:
+            observation (torch.Tensor): Observation from the environment.
+
+        Returns:
+             torch.Tensor: Calculated entropy values given observation
+
+        """
+        dist = self.forward(observation)
+        return dist.entropy()
+
+    def reset(self, dones=None):
+        """Reset the environment.
+
+        Args:
+            dones (numpy.ndarray): Reset values
```

Comment:
please specify shapes.

also, what is `dones`?? isn't that something an environment produces?

Reply:
Oh yeah. This is perhaps the most confusing parameter of the vectorized policy interface. It's called `dones` everywhere else, so I'm not really sure we should change it here. Having said that, it's only passed in a few places, and never by keyword.

It's certainly not a blocking detail.

Reply:
thanks @krzentner .

this doc still needs a shape.